File: Computer/Ram.go
package Computer

import (
	"changeme/utils"
	"github.com/shirou/gopsutil/mem"
)

type RamInfo struct {
	Total float64 `json:"total"`
	Used  float64 `json:"used"`
	UsePr float64 `json:"usePr"`
}

type RAM struct {
	//内存名
	Name string
	//总容量
	Total string
	//上市时间
	CreateTime string
	//时序
	CL string
	//频率
	MHZ string
	//价格
	Price float64
	//图片地址
	ImgUrl string
	//内存类型
	Type string
}

// GetRamInfo
// @Description: 获取本机内存信息
// @return RamInfo
func GetRamInfo() RamInfo {
	m := RamInfo{}
	ramInfo, _ := mem.VirtualMemory()
	m.Total = utils.BToGb(ramInfo.Total)
	m.Used = utils.BToGb(ramInfo.Used)
	m.UsePr = ramInfo.UsedPercent
	return m
}

func GetRamList() {

}

File: frontend/src/views/PCHardware/RAM.js
import { ref } from "vue";

const RAM =()=>{
const RAMInfo=ref([
  {label:"内存型号",val:"",key:"Name"},
  {label:"内存容量",val:"",key:"Total"},
  {label:"时序",val:"",key:"CL"},
  {label:"频率",val:"",key:"MHZ"},
  {label:"颗粒",val:"",key:"Type"},
  {label:"首发价格",val:"",key:"Price"},
  {label:"图片",val:"",key:"ImgUrl"},
  {label:"上市日期",val:"",key:"CreateTime"},
])
return {
  RAMInfo
}
}
export default RAM;